api/lib/mno_enterprise/concerns/controllers/jpi/v1/product_instances_controller.rb

```diff
@@ -0,0 +1,50 @@
+module MnoEnterprise::Concerns::Controllers::Jpi::V1::ProductInstancesController
+  extend ActiveSupport::Concern
+
+  #==================================================================
+  # Included methods
+  #==================================================================
+  # 'included do' causes the included code to be evaluated in the
+  # context where it is included rather than being executed in the module's context
+  included do
+    DEPENDENCIES = [:app_instance, :'app_instance.app', :product,
+                    :'product.values', :'product.values.field']
+    respond_to :json
+  end
+
+  #==================================================================
+  # Instance methods
+  #==================================================================
+  # GET /mnoe/jpi/v1/organization/1/product_instances
+  def index
+    statuses = MnoEnterprise::ProductInstance::ACTIVE_STATUSES.join(',')
+    @product_instances = MnoEnterprise::ProductInstance.includes(*DEPENDENCIES).where(organization_id: parent_organization.id, 'status.in': statuses, 'fulfilled_only': true).to_a.select do |i|
+      can?(:access_product_instance, i)
+    end
+  end
+
+  # TODO: Copied over structure for AppInstances.
+  #       Needs to be setup for ProductInstances when doing complete migration
+  #       from AppInstance to ProductInstance
+
+  # POST /mnoe/jpi/v1/organization/1/product_instances
```

Comment:
We should not support the create/destroy actions anyway as products will be managed via subscriptions